src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -636,46 +643,37 @@ internal static IDataTransform Create(IHostEnvironment env, Options args, IDataV
         private sealed class Transformer : ITransformer
         {
             private const string TransformDirTemplate = "Step_{0:000}";
+            private const uint VerIDataTransform = 0x00010001;
 
             private readonly IHost _host;
-            private readonly IDataView _xf;
+            private readonly TransformerChain<ITransformer> _chain;
 
-            internal Transformer(IHostEnvironment env, IDataView input, IDataView view)
+            internal Transformer(IHostEnvironment env, TransformerChain<ITransformer> chain)
             {
+                Contracts.AssertValue(env);
+                env.AssertValue(chain);
                 _host = env.Register(nameof(Transformer));
-                _xf = ApplyTransformUtils.ApplyAllTransformsToData(_host, view, new EmptyDataView(_host, input.Schema), input);
+                _chain = chain;
             }
 
             public DataViewSchema GetOutputSchema(DataViewSchema inputSchema)
             {
                 _host.CheckValue(inputSchema, nameof(inputSchema));
-                return Transform(new EmptyDataView(_host, inputSchema)).Schema;
+                return _chain.GetOutputSchema(inputSchema);
             }
 
             public IDataView Transform(IDataView input)
             {
                 _host.CheckValue(input, nameof(input));
-                return ApplyTransformUtils.ApplyAllTransformsToData(_host, _xf, input);
+                return _chain.Transform(input);
             }
 
             bool ITransformer.IsRowToRowMapper => true;
```

Comment:
>true [](start = 50, length = 4)

If we're going to do that, that's fine, but it might be best to have an assert that the chain is a row to row mapper in the constructor then. (Only if you have to make another change for some other reason anyway.)